src/Microsoft.AspNet.Mvc.ModelBinding/Binders/MutableObjectModelBinder.cs

```diff
@@ -480,7 +480,7 @@ internal void ProcessDto(ModelBindingContext bindingContext, ComplexModelDto dto
             {
                 try
                 {
-                    property.SetValue(bindingContext.Model, value);
+                    propertyMetadata.PropertySetter(bindingContext.Model, value);
```

Comment:
There's code on Line 430 that does reflection to fetch all properties. Could you file a follow up work item to remove using reflection and use model metadata?